src/command.py

```diff
@@ -78,15 +78,33 @@ def cursorless_multiple_target_command(
         arg3: any = NotSet,
     ):
         """Execute multi-target cursorless command"""
-        args = list(filter(lambda x: x is not NotSet, [arg1, arg2, arg3]))
         actions.user.vscode_with_plugin_and_wait(
             "cursorless.command",
-            get_spoken_form(),
-            action,
-            targets,
-            *args,
+            construct_cursorless_command_argument(
+                action=action,
+                targets=targets,
+                args=[x for x in [arg1, arg2, arg3] if x is not NotSet],
+            ),
         )
 
 
+def construct_cursorless_command_argument(
+    action: str, targets: list[dict], args: list[any]
+):
+    try:
+        use_pre_phrase_snapshot = actions.user.did_emit_pre_phrase_signal()
+    except KeyError:
+        use_pre_phrase_snapshot = False
+
+    return {
+        "version": 1,
+        "spokenForm": get_spoken_form(),
+        "action": action,
+        "targets": targets,
+        "extraArgs": args,
```

Comment:
extraArgs I'm not sold on this name. May be just `arguments`?

Reply:
Yeah thought a bit about this one. They're "extra" in the sense that targets are also arguments to the action. Something does smell a bit weird here. But I think it's prob a symptom of the fact that targets have such special treatment relative to other args to the action. Idk

Reply:
I get what you're coming from. I can live with extraArgs